From dc0b67adb08b4fb68b63c38ef286b19ffb8aaed5 Mon Sep 17 00:00:00 2001
From: Assam Boudjelthia <assam.boudjelthia@qt.io>
Date: Thu, 21 Aug 2025 15:45:58 +0300
Subject: [PATCH 205/553] AndroidTestRunner: don't print crashes by default

In most cases after a crash, logs will be too long and will
flood the test output, making it not ideal and mainly unexpected.
Instead of printing the crash logs and logcat by default, have
it done only when it's explicitly requested via --show-logcat.

For CI though, add the --show-logcat by default so we don't lose
that verbosity.

Change-Id: I2306b97378b6218ed22ba98aa50deefa9c9cd968
Reviewed-by: Joerg Bornemann <joerg.bornemann@qt.io>
(cherry picked from commit a67f47e8cad66749926ed172766c790f04a661e1)
---
 cmake/QtTestHelpers.cmake            | 4 ++++
 src/tools/androidtestrunner/main.cpp | 6 +++---
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/cmake/QtTestHelpers.cmake b/cmake/QtTestHelpers.cmake
index cf8a085b26d..e08f81eab2d 100644
--- a/cmake/QtTestHelpers.cmake
+++ b/cmake/QtTestHelpers.cmake
@@ -699,6 +699,10 @@ function(qt_internal_add_test name)
             list(APPEND extra_test_args "--verbose")
         endif()
 
+        if(build_environment STREQUAL "ci")
+            list(APPEND extra_test_args "--show-logcat")
+        endif()
+
         if(arg_ANDROID_TESTRUNNER_PRE_TEST_ADB_COMMANDS)
             foreach(command IN LISTS arg_ANDROID_TESTRUNNER_PRE_TEST_ADB_COMMANDS)
                 list(APPEND extra_test_args "--pre-test-adb-command" "${command}")
diff --git a/src/tools/androidtestrunner/main.cpp b/src/tools/androidtestrunner/main.cpp
index eb4c84fbd75..5e3e4c3c6dc 100644
--- a/src/tools/androidtestrunner/main.cpp
+++ b/src/tools/androidtestrunner/main.cpp
@@ -251,7 +251,6 @@ static void printHelp()
                     "\n"
                     "    --show-logcat: Print Logcat output to stdout. If an ANR occurs during\n"
                     "       the test run, logs from the system_server process are included.\n"
-                    "       This argument is implied if a test crashes.\n"
                     "\n"
                     "    --ndk-stack: Path to ndk-stack tool that symbolizes crash stacktraces.\n"
                     "       By default, ANDROID_NDK_ROOT env var is used to deduce the tool path.\n"
@@ -744,7 +743,7 @@ void analyseLogcat(const QString &timeStamp, int *exitCode)
     // If we have a crash, attempt to print both logcat and the crash buffer which
     // includes the crash stacktrace that is not included in the default logcat.
     const bool testCrashed = *exitCode == EXIT_ERROR && !g_testInfo.isTestRunnerInterrupted.load();
-    if (g_options.showLogcatOutput || testCrashed) {
+    if (testCrashed) {
         qDebug() << "********** logcat dump **********";
         qDebug().noquote() << testLogcat.join(u'\n').trimmed();
         qDebug() << "********** End logcat dump **********";
@@ -922,7 +921,8 @@ int main(int argc, char *argv[])
 
     int exitCode = testExitCode();
 
-    analyseLogcat(formattedStartTime, &exitCode);
+    if (g_options.showLogcatOutput)
+        analyseLogcat(formattedStartTime, &exitCode);
 
     exitCode = pullResults() ? exitCode : EXIT_ERROR;
 
-- 
2.51.2

